### PR TITLE
feat(s3): Automatically discover bucket location

### PR DESCRIPTION
**Pull request by bbernays**
#### Summary

Fixes https://github.com/cloudquery/cloudquery/issues/6792

This PR sets a `DefaultRegion` for making the `GetBucketLocation`... This allows a user to specify a region via environment variables or profile which will override this value

### CODE REVIEW

File: plugins/destination/s3/client/client.go [hunk 1]
```diff
@@ -54,11 +54,16 @@ func New(ctx context.Context, logger zerolog.Logger, spec specs.Destination) (de
 	}
 	c.pluginSpec.SetDefaults()
 
-	cfg, err := config.LoadDefaultConfig(ctx)
+	cfg, err := config.LoadDefaultConfig(ctx, config.WithRegion("us-east-1"))
```
Comment (bbernays) [inline plugins/destination/s3/client/client.go:57]:
```suggestion
	cfg, err := config.LoadDefaultConfig(ctx, config.WithDefaultRegion("us-east-1"))
```

**Review by yevgenypats — COMMENTED** [review]
Looks good. Question: permission wise: is it possible someone who's using the S3 plugin wont have permissions to do `getBucketLocation` ?

**Review by candiduslynx — APPROVED** [review]

**Comment by bbernays** [comment]
> Looks good. Question: permission wise: is it possible someone who's using the S3 plugin wont have permissions to do `getBucketLocation` ?

I don't see it as an issue... But I will document the IAM permissions required as part of this PR

**Review by yevgenypats — APPROVED** [review]
LGTM.

**Pull request merged**